src/sync.ts

```diff
@@ -69,14 +70,32 @@ const BUFFER_PERIOD_MS = 80 * 1000;
 const FAILED_SYNC_ERROR_THRESHOLD = 3;
 
 export enum SyncState {
+    /** Emitted after we try to sync more than `FAILED_SYNC_ERROR_THRESHOLD`
+     * times and are still failing. Or when we enounter a hard error like the
+     * token being invalid. */
     Error = "ERROR",
+    /** Emitted after the first sync events are ready (this could even be sync
+     * events from the cache) */
     Prepared = "PREPARED",
+    /** Emitted when the sync loop is no longer running */
     Stopped = "STOPPED",
+    /** Emitted after each sync request happens */
     Syncing = "SYNCING",
+    /** Emitted after a connectivity error and we're ready to start syncing again */
     Catchup = "CATCHUP",
+    /** Emitted for each time we try reconnecting. Will switch to `Error` after
+     * we reach the `FAILED_SYNC_ERROR_THRESHOLD`
+     */
     Reconnecting = "RECONNECTING",
 }
 
+// Room versions where "insertion", "batch", and "marker" events are controlled
+// by power-levels. MSC2716 is supported in existing room versions but they
+// should only have special meaning when the room creator sends them.
+const MSC2716_ROOM_VERSIONS = [
+    'org.matrix.msc2716v3',
+];
```

Comment:
This should really be signalled via the `m.room_versions` capability

Reply:
Good shout 👍 Will create a Synapse PR to add it

Reply:
Synapse PR: https://github.com/matrix-org/synapse/pull/12516

Reply:
I think we may have to opt for this static list for now. The consensus on the PR is not to stack the MSC behaviors, see https://github.com/matrix-org/synapse/pull/12516#pullrequestreview-948231079 ([MSC3244](https://github.com/matrix-org/matrix-doc/pull/3244) to add `room_capabilities` hasn't landed yet)